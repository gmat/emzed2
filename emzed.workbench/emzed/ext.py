import pkg_resources

for ep in pkg_resources.iter_entry_points("emzed_plugin", name="package"):
    print ep
    package = ep.load()
    exec "%s=package" % package.__name__
    del package
    del ep
del pkg_resources

