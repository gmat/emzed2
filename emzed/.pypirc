[distutils]
index-servers = 
  local

[local]
repository: http://127.0.0.1:8080
username:uschmitt
password:pillepalle

