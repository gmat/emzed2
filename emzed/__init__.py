# encoding:latin-1

from version import version as __version__


