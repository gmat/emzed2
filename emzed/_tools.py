
def is_started_from_cmdline():

    import inspect
    info = inspect.stack()[-1][1]
    result = False
    if info != "<string>":
        if info.endswith("emzed.workbench"):
            result = True
    return result


def gui_running():
    from PyQt4.QtGui import QApplication
    return QApplication.instance() is not None


def is_first_start():
    from emzed.core.config import _UserConfig
    import os
    return not os.path.exists(_UserConfig.config_file_path())
