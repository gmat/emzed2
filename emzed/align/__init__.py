from rt_align import rtAlign
from mz_align import mzAlign
