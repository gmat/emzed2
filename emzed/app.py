
import pkg_resources
reload(pkg_resources)  # migh be loaded already and is not up to date

for ep in pkg_resources.iter_entry_points("emzed_package", name="main"):
    runner = ep.load()
    print "$$$ loaded", runner
    exec "%s=runner" % ep.module_name.split(".")[0]
    del ep
    del runner

del pkg_resources

