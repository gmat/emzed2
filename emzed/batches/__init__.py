from peak_picker import *

# remove clutter
try:
    del peak_picker
except: 
    pass
