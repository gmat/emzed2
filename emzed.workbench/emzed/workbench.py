def run():
    print 42

def main():
    import os
    print os.system("pip install -i http://127.0.0.1:3142/root/dev/+simple emzed.core")
    import emzed.core
    print emzed.core



