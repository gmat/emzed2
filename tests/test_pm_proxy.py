# encoding: utf-8
from __future__ import print_function


import emzed


def test_0(path):
    from emzed.core.data_types.ms_types import PeakMapProxy
    pm = PeakMapProxy(path("data/SHORT_MS2_FILE.mzData"))
    # this will trigger loading:
    n = len(pm)
    assert n == 41


def test_1(path, tmpdir):
    from emzed.core.data_types.ms_types import PeakMapProxy
    pm = emzed.io.loadPeakMap(path("data/SHORT_MS2_FILE.mzData"))
    t = emzed.utils.toTable("id", (1, 2, 3), type_=int)
    t.addColumn("peakmap", pm, type_=object)
    t.store(tmpdir.join("without_comp.table").strpath, True)
    t.store(tmpdir.join("with_comp.table").strpath, True, True, peakmap_cache_folder=tmpdir.strpath)

    assert isinstance(t.peakmap.uniqueValue(), PeakMapProxy)
    assert not t.peakmap.uniqueValue()._loaded
    assert t.peakmap.uniqueValue().uniqueId() is not None    # must not trigger loading !
    assert not t.peakmap.uniqueValue()._loaded

    tn = emzed.io.loadTable(tmpdir.join("with_comp.table").strpath)
    pm = tn.peakmap.uniqueValue()
    assert t.peakmap.uniqueValue().uniqueId() is not None    # must not trigger loading !
    assert not tn.peakmap.uniqueValue()._loaded
    assert len(pm) == 41  # triggers loading
    assert tn.peakmap.uniqueValue()._loaded

    emzed.io.storeTable(t, tmpdir.join("with_comp_2.table").strpath, True, True, tmpdir.strpath)

    tn = emzed.io.loadTable(tmpdir.join("with_comp_2.table").strpath)
    pm = tn.peakmap.uniqueValue()
    assert isinstance(pm, PeakMapProxy)
    assert t.peakmap.uniqueValue().uniqueId() is not None    # must not trigger loading !
    assert not tn.peakmap.uniqueValue()._loaded
    assert len(pm) == 41  # triggers loading
    assert tn.peakmap.uniqueValue()._loaded


def test_squeeze(path):

    from emzed.core.data_types.ms_types import PeakMapProxy
    pm = PeakMapProxy(path("data/SHORT_MS2_FILE.mzData"))
    # this will trigger loading:
    n = len(pm)
    assert n == 41
    assert isinstance(pm, PeakMapProxy)
    assert "spectra" in pm.__dict__

    pm.squeeze()
    assert "spectra" not in pm.__dict__
    assert len(pm) == 41  # triggers loading
    assert "spectra" in pm.__dict__
