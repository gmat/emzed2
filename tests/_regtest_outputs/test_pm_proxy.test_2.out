['peak_map_009bf7c2dd290e7f183de61545bb2ec8eb349286bbe1136852250fb5c7bf6157.pickle', 'with_comp.table']
41
41
id       peakmap
int      object
------   ------
1        <emzed.core.data_types.ms_types.PeakMapProxy object at 0x?????????>
2        <emzed.core.data_types.ms_types.PeakMapProxy object at 0x?????????>
3        <emzed.core.data_types.ms_types.PeakMapProxy object at 0x?????????>

41
id       peakmap
int      object
------   ------
1        <emzed.core.data_types.ms_types.PeakMapProxy object at 0x?????????>
2        <emzed.core.data_types.ms_types.PeakMapProxy object at 0x?????????>
3        <emzed.core.data_types.ms_types.PeakMapProxy object at 0x?????????>

